src/nbl/asset/interchange/CPLYMeshWriter.cpp

```diff
@@ -85,37 +85,84 @@ bool CPLYMeshWriter::writeAsset(io::IWriteFile* _file, const SAssetWriteParams&
 
     const asset::E_WRITER_FLAGS flags = _override->getAssetWritingFlags(ctx, mesh, 0u);
 
+    auto getConvertedMeshBufferWithNewIndexBuffer = [&]() -> core::smart_refctd_ptr<asset::ICPUMeshBuffer>
+    {
+        auto inputMeshBuffer = *meshbuffers.begin();
+
+        const auto inputIndexType = inputMeshBuffer->getIndexType();
+        const auto inputPrimitiveType = inputMeshBuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType;
+
+        const void* inputIndices = inputMeshBuffer->getIndices();
+        auto inputIndexCount = inputMeshBuffer->getIndexCount();
+
+        if (inputMeshBuffer->getIndexBufferBinding().buffer && inputPrimitiveType != asset::EPT_TRIANGLE_LIST)
+        {
+            auto copyMeshBuffer = core::smart_refctd_ptr_static_cast<asset::ICPUMeshBuffer>(inputMeshBuffer->clone());
+
+            if (inputPrimitiveType == asset::EPT_TRIANGLE_FAN || inputPrimitiveType == asset::EPT_TRIANGLE_STRIP)
+            {
+                core::smart_refctd_ptr<ICPUBuffer> newIndexBuffer;
+
+                if (inputPrimitiveType == asset::EPT_TRIANGLE_FAN)
+                    newIndexBuffer = IMeshManipulator::idxBufferFromTrianglesFanToTriangles(inputIndices, inputIndexCount, inputIndexType, inputIndexType);
+                else if (inputPrimitiveType == asset::EPT_TRIANGLE_STRIP)
+                    newIndexBuffer = IMeshManipulator::idxBufferFromTriangleStripsToTriangles(inputIndices, inputIndexCount, inputIndexType, inputIndexType);
+
+                asset::SBufferBinding<asset::ICPUBuffer> indexBufferBinding;
+                indexBufferBinding.buffer = newIndexBuffer;
+                indexBufferBinding.offset = 0u;
+
+                copyMeshBuffer->setIndexBufferBinding(std::move(indexBufferBinding));
+                return copyMeshBuffer;
+            }
+        }
+        else 
+            return nullptr;
+    };
+
+    const auto cpuConvertedMeshBuffer = getConvertedMeshBufferWithNewIndexBuffer();
+    const asset::ICPUMeshBuffer* rawConvertedMeshBuffer = cpuConvertedMeshBuffer.get() ? cpuConvertedMeshBuffer.get() : *meshbuffers.begin();
+    const auto primitiveType = rawConvertedMeshBuffer->getPipeline()->getPrimitiveAssemblyParams().primitiveType;
+    const auto indexType = rawConvertedMeshBuffer->getIndexType();
+    uint32_t faceCount = {}; 
+    size_t vertexCount = {};
+
+    void* indices = nullptr;
+    {
+        auto indexCount = rawConvertedMeshBuffer->getIndexCount();
+
+        indices = _NBL_ALIGNED_MALLOC(indexCount * (indexType == asset::EIT_16BIT ? 2u : 4u), _NBL_SIMD_ALIGNMENT);
+        memcpy(indices, rawConvertedMeshBuffer->getIndices(), indexCount * (indexType == asset::EIT_16BIT ? 2u : 4u));
```

Comment:
why are you copying the indices?

Reply:
I wanted to just raw point to it, but the writing function takes non const pointer somehow and it modifies the indicies under `_forceFaces` bool condition, I will leave it as it is I think

Reply:
not my implementation, but it is needed probably

Reply:
that's why it was copied before I assume